### PR TITLE
Components: update CheckboxControl readme

### PR DESCRIPTION
**Pull request by davewhitley**
## Description

This PR adds design guideline documentation in addition to the development documentation that existed previously. These guidelines are 'best practices' for the usage of the component, as well as describing the component in more detail.

## Screenshots

A preview can be seen here, although it doesn't represent what it will look like in the handbook:

[preview](https://codepen.io/drw158/pen/MxwxXO)

## Feedback

I'd appreciate feedback on:

- Whether my markdown syntax is correct
- Design guidelines
- Double-checking a11y of the document
- Seeing if the design guidelines align with the development documentation

Thank you!

### CODE REVIEW

**Comment by drw158** [comment]
Sorry about all of the requested reviews. I didn't do that so it must be an automated thing?

**Comment by drw158** [comment]
The images do not have alt description because if the images were removed, the content would not lose any value (except for maybe the first image?). 

ref:(https://github.com/WordPress/gutenberg/pull/11226#issuecomment-437421535)

A11y feedback on this would be great.

**Comment by aduth** [comment]
>Sorry about all of the requested reviews. I didn't do that so it must be an automated thing?

It's part of the code owners workflow implemented in the project. All those assigned had volunteered themselves to be assigned.

See also:

- https://help.github.com/en/articles/about-code-owners
- https://github.com/WordPress/gutenberg/blob/master/.github/CODEOWNERS
- https://github.com/WordPress/gutenberg/pull/13604

File: packages/components/src/checkbox-control/README.md
Comment (aduth) [inline packages/components/src/checkbox-control/README.md]: We lose some useful/standards-compliant tabbing here.

Comment (drw158) [reply packages/components/src/checkbox-control/README.md]: I believe this is now fixed. Do you mind checking? If

**Comment by drw158** [comment]
@karmatosed This PR does not have a design or a11y review, but I think it should be good to merge once an owner approves.

**Comment by karmatosed** [comment]
@drw158 I agree and I will now approve. We can (and should) iterate, but it's important to get these in.

**Review by karmatosed — APPROVED** [review]

File: packages/components/src/checkbox-control/README.md
Comment (aduth) [reply packages/components/src/checkbox-control/README.md]: > I believe this is now fixed. Do you mind checking? If

Looks fine now. Thanks 👍

**Pull request merged**
